### PR TITLE
PKG add libheif, pyheif, and pillow_heif (decoder only)

### PR DESCRIPTION
**Pull request by ryanking13**
### Description

Close: #2785 

This adds libheif and Python packages that use libheif to support HEIC image format.

Note that this is "decoder" only. libheif use libde265 for decoder and libx265 for encoder, and I only added libde265 in this PR.
I think decoding is more important in our use case, hence this PR.

For those who are interested in "encoding" heic format images (possibly me in the future...), I was able to build libx265 but Pyodide hangs when I tried to use it via pillow_heif. You can see [meta.yaml file of libx265 that I tried here](https://github.com/ryanking13/pyodide/blob/pkg/heif/packages/libx265/meta.yaml).

cc: @teonbrooks

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

File: packages/pillow_heif/test_pillow_heif.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            assert heif_file.mode == "RGBA"
+            assert len(heif_file.data) == 278784
+            assert heif_file.stride == 1056
+        """
+    )
+
+
+def test_pillow(selenium):
+    selenium.load_package(["Pillow", "pillow_heif"])
+    selenium.run(
```
Comment (rth) [inline packages/pillow_heif/test_pillow_heif.py:31]: `@run_in_pyodide(packages = ["Pillow", "pillow_heif"])` here and above ?

**Review by rth — APPROVED** [review]
Thanks @ryanking13 ! We could maybe use `run_in_pyodide` for tests, but otherwise LGTM.

Could you please restart the benchmark CI job to make sure that the failure is unrelated?

**Comment by ryanking13** [comment]
> Thanks @ryanking13 ! We could maybe use `run_in_pyodide` for tests, but otherwise LGTM.
> 
> Could you please restart the benchmark CI job to make sure that the failure is unrelated?

Done. Thanks for the review!

**Comment by teonbrooks** [comment]
thanks @ryanking13!

**Pull request merged**